examples/vega-plotting.dx

```diff
@@ -0,0 +1,364 @@
+         show = (\ x.
+              case x of
+                   Area -> "area"
+                   Bar -> "bar"
+                   Circle -> "circle"
+                   Line -> "line"
+                   Point -> "point"
+                   Rect -> "rect"
+                   Rule-> "rule"
+                   Square -> "square"
+                   Tick -> "tick")
+
+' Most things in VL can take in extra visual options.
+  To avoid specifying these, we will take in as
+  JSON.
+
+data Opts a =
+     WithOpts a Value
+
+def pure (x:a) : Opts a =
+    WithOpts x AsNone
+    
+def pureLs (x:a) : List (Opts a) =
+   AsList 1 [WithOpts x AsNone]
+
+def mergeOpts [ToJSON a, ToJSON b] (x : a) (y : b) : Value =
+     case toJSON x of 
+         (AsObject x') -> case toJSON y of
+                  (AsObject y') -> AsObject $ x' <> y'
+                  (AsNone) -> AsObject x'
+         (AsNone) -> toJSON y
+
+
+
+data VLChart row col v =
+     AsVLDescriptor (Opts Mark) v (col => ({title: Header &
+                              encType: EncodingType &
+                              encodings: List (Opts Channel) &
+                              rows: row => Value
+                              })) 
+
+instance [ToJSON v] ToJSON (VLChart r c v)
+   toJSON = \ x.
+    (AsVLDescriptor mark opts df) = x
+    -- Make the mark
+    (WithOpts mtype options) = mark
+    jmark = ("mark", mergeOpts options [("type", show mtype)])
+
+    -- Make the data
+    jdf = toJSON $ for row : r. toJSON $ for col : c.
+                   ("col" <> (show $ ordinal col),
+                    toJSON (getAt #rows df.col).row)
+    jdata = ("data", toJSON [("values", jdf)])
+
+    -- Make the encodings
+    jencodings = toJSON $ concat $ for col : c.
+        (AsList v encopts) = getAt #encodings df.col
+        AsList v $ for f. 
+           (WithOpts channel encoptions) = encopts.f
+           (show channel,
+            mergeOpts encoptions 
+            [
+              ("field", "col" <> (show $ ordinal col)),
+              ("type",  show $ getAt #encType df.col),
+              ("title", getAt #title df.col)
+              ])
+    jencode = ("encoding", jencodings)
+    mergeOpts opts [jdata, jmark, jencode]
+    
+
+def showVega (x: Value) : String =
+    "<iframe width=\"100%\" frameborder=\"0\" scrolling=\"no\" onload=\"resize_iframe(this)\" srcdoc='<html> 
+    <head><script src=\"https://cdn.jsdelivr.net/npm/vega@5.17.0\"></script>  
+          <script src=\"https://cdn.jsdelivr.net/npm/vega-lite@4.17.0\"></script>  
+          <script src=\"https://cdn.jsdelivr.net/npm/vega-embed@6.12.2\"></script> 
+    </head> 
+    <body> 
+           <div id=\"vis\"></div>
+           <script>vegaEmbed(\"#vis\"," <> (show x) <> ");</script>
+    </body>
+    </html>'>
+    </iframe>"
+
+' ## Example: Bar Chart
+
+' Start with a well type and useful Dex record
+
+df1 = {a = ["A", "B", "C", "D", "E", "F", "G", "H", "I"],
+       b = [28, 55, 43, 91, 81, 53, 19, 87, 52]}
```

Comment:
Any reason to make data frames records-of-tables? I thought that table-of-records is a bit more intuitive, because that's closer to relational algebra

Reply:
Makes sense. Switched to rows.